docs/security.md

```diff
@@ -0,0 +1,121 @@
+passed to the Collector MUST be validated prior to be loaded. If an invalid
+configuration is detected, the Collector MUST fail to start as a protective
+mechanism.
+
+The configuration drives the Collector's behavior and care should be taken to
+ensure the configuration only enables the minimum set of capabilities and as
+such exposes the minimum set of required ports. In addition, any incoming or
+outgoing communication SHOULD leverage TLS and/or authentication.
+
+The Collector keeps the configuration in memory, but where the configuration is
+loaded from at start time depends on the packaging used. For example, in
+Kubernetes secrets and configmaps CAN be leveraged. In comparison, the Docker
+image embeds the configuration in the container where is it not stored in an
+encrypted manner by default.
+
+The configuration MAY contain sensitive information including:
+
+- Authentication information such as API tokens
+- TLS certificates including private keys
+
+Sensitive information SHOULD be stored securely such as on an encrypted
+filesystem or secret store. Environment variable CAN be used to handle
+sensitive and non-sensitive data as the Collector MUST support environment
+variable expansion.
+
+More information about configuration is provided in the following sections.
+
+## Permissions
+
+The Collector supports running as a custom user and SHOULD NOT be run as a
+root/admin user. For the majority of use-cases, the Collector SHOULD NOT require
+privileged access to function. Some components MAY require privileged access
+and care should be taken before enabling these components. Collector components
+MAY require external permissions including network access or RBAC access.
+
+More information about permissions is provided in the following sections.
+
+## Receivers and Exporters
+
+Receivers and Exporters can be either push or pull-based. In either case, the
+connection established SHOULD be over a secure and authenticated channel.
+Unused receivers and exporters SHOULD be disabled to minimize the attack vector
+of the Collector. Receivers and exporters MUST default to TLS connections (via
+the `insecure: False` configuration setting).
+
+Receivers and Exporters MAY expose buffer, queue, payload, and/or worker
+settings. If these settings are available, care should be taken before
+modifying the default values. Improperly setting these values may expose the
+Collector to additional attack vectors including resource exhaustion.
+
+> It is possible that a receiver MAY require the Collector run in a privileged
+> mode in order to operate, which could be a security concern, but today this
+> is not the case.
+
+## Processors
+
+Processors sit between receivers and exporters. They are responsible for
+processing the data in some way. From a security perspective, they are useful
+in a couple ways.
+
+### Scrubbing sensitive data
+
+It is common for a Collector to be used to scrub sensitive data before
```

Comment:
Just noting here that scrubbing or obfuscation processors for spans with sensitive information in attributes (such as `db.statement`: https://github.com/open-telemetry/opentelemetry-specification/blob/1afab39e5658f807315abf2f3256809293bfd421/specification/trace/semantic_conventions/database.md#call-level-attributes) are going to become an increasingly common request and/or use case.  We may want to consider where having default obfuscation or scrubbing of known sensitive attributes lands on the roadmap

Reply:
True, let me add some words around this